src/server/editorServices.ts

```diff
@@ -0,0 +1,1646 @@
+            return this.projectFilename;
+        }
+
+        // add a root file to project
+        addRoot(info: ScriptInfo) {
+            info.defaultProject = this;
+            return this.compilerService.host.addRoot(info);
+        }
+
+        filesToString() {
+            var strBuilder = "";
+            ts.forEachValue(this.filenameToSourceFile,
+                sourceFile => { strBuilder += sourceFile.fileName + "\n"; });
+            return strBuilder;
+        }
+
+        setProjectOptions(projectOptions: ProjectOptions) {
+            this.projectOptions = projectOptions;
+            if (projectOptions.compilerOptions) {
+                this.compilerService.setCompilerOptions(projectOptions.compilerOptions);
+            }
+            // TODO: format code options
+        }
+    }
+
+    interface ProjectOpenResult {
+        success?: boolean;
+        errorMsg?: string;
+        project?: Project;
+    }
+
+    function copyListRemovingItem<T>(item: T, list: T[]) {
+        var copiedList: T[] = [];
+        for (var i = 0, len = list.length; i < len; i++) {
+            if (list[i] != item) {
+                copiedList.push(list[i]);
+            }
+        }
+        return copiedList;
+    }
+
+    interface ProjectServiceEventHandler {
+        (eventName: string, project: Project): void;
+    }
+
+    export class ProjectService {
+        filenameToScriptInfo: ts.Map<ScriptInfo> = {};
+        // open, non-configured files in two lists
+        openFileRoots: ScriptInfo[] = [];
+        openFilesReferenced: ScriptInfo[] = [];
+        // projects covering open files
+        inferredProjects: Project[] = [];
+
+        constructor(public host: ServerHost, public psLogger: Logger, public eventHandler?: ProjectServiceEventHandler) {
+            // ts.disableIncrementalParsing = true;
+        }
+
+        watchedFileChanged(fileName: string) {
+            var info = this.filenameToScriptInfo[fileName];
+            if (!info) {
+                this.psLogger.info("Error: got watch notification for unknown file: " + fileName);
+            }
+
+            if (!this.host.fileExists(fileName)) {
+                // File was deleted
+                this.fileDeletedInFilesystem(info);
+            }
+            else {
+                if (info && (!info.isOpen)) {
+                    info.svc.reloadFromFile(info.fileName);
+                }
+            }
+        }
+        
+
+        log(msg: string, type = "Err") {
+            this.psLogger.msg(msg, type);
+        }
+
+        closeLog() {
+            this.psLogger.close();
+        }
+
+        createInferredProject(root: ScriptInfo) {
+            var iproj = new Project(this);
+            iproj.addRoot(root);
+            iproj.finishGraph();
+            this.inferredProjects.push(iproj);
+            return iproj;
+        }
+
+        fileDeletedInFilesystem(info: ScriptInfo) {
+            this.psLogger.info(info.fileName + " deleted");

```

Comment:
This is misleading, it implies that the service deleted it; it sounds more like you *detected* a file getting deleted